Content.Server/Stunnable/Systems/StunbatonSystem.cs

```diff
@@ -130,5 +144,22 @@ private void SendPowerPulse(EntityUid target, EntityUid? user, EntityUid used)
                 User = user
             }, false);
         }
+
+        private void OnSolutionChanged(EntityUid uid, StunbatonComponent component, SolutionChangedEvent args)
+        {
+            if (TryComp<BatteryComponent>(uid, out var battery))
+            {
+                _riggedSystem.IsRigged(uid, component, battery.SolutionName, args);
+            }
+
+            if (TryComp<RiggedComponent>(uid, out var rig) && rig.IsRigged)
+            {
+                _adminLogger.Add(LogType.Explosion, LogImpact.Medium, $"{ToPrettyString(uid)} has been rigged up to explode when used.");
+
+                // Go ahead - inject plasma into an enabled baton.
+                if (component.Activated)
+                    _riggedSystem.Explode(uid, battery, uid);
+            }
+        }
```

Comment:
As above just remove all of this except for component.Activated and slap a TODO to remove because people keep slapping more code into stunbaton when it needs splitting out into 3 components.

Reply:
Why not just check if it's a stunbaton getting rigged with

```C#
            // Check if its the stun baton getting rigged
            if (!TryComp<StunbatonComponent>(uid, out var stunbatonComponent))
                return;

            // If it is AND is activated - explode it
            if (stunbatonComponent.Activated)
                Explode(uid, battery, uid);
                ```